File: online-judge-solutions/Codeforces/1629A.py
#https://codeforces.com/problemset/problem/1629/A
#Lang		:	Python 3.8
#Time		:	62 ms
#Memory         :	0 KB
for _ in range(int(input())):
    n,k=map(int,input().split())
    a=list(map(int,input().split()))[:n]
    b=list(map(int,input().split()))[:n]
    p=[]
    for i in range(n):p.append([a[i],b[i]])
    p.sort()
    for i in range(n):
        if(p[i][0]<=k): k+=p[i][1]
        else: break
    print(k)

File: online-judge-solutions/Codeforces/1629A.cpp
//https://codeforces.com/problemset/problem/1629/A
//Lang		:	GNU C++14
//Time		:	15 ms
//Memory	:	100 KB
#include<iostream>
#include<vector>
#include<utility>
#include<algorithm>
using namespace std;
#define fastInp cin.tie(0); cout.tie(0); ios_base::sync_with_stdio(0);
int main(){
	fastInp;
	int t,n,k;
	for(cin>>t;t--;){
		cin>>n>>k;
		vector<int> a(n),b(n);
		for(auto &i:a){cin>>i;}
		for(auto &i:b){cin>>i;}
		vector<pair<int,int>> p;
		for(int i=0;i<n;i++){p.push_back(make_pair(a[i],b[i]));}
		sort(begin(p),end(p));
		int s=k;
		for(int i=0;i<n;i++){
			if(p[i].first<=s) s+=p[i].second;
			else break;
		}
		cout<<s<<endl;
	}
	return 0;
}
